File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> dict;
        for(auto c:s) {
            dict[c]++;
        }
        for(auto c:t){
            dict[c]--;
        }
        for(auto i:dict) {
            if(i.second != 0) {
                return false;
            }
        }
        return true;

    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        word = dict()
        if len(s) != len(t) :
            return False
        for index, character in enumerate (s) :
            if character in word : 
                word[character] += 1
            else :
                word[character] = 1
        for index, character in enumerate (t) :
            if character in word : 
                word[character] -= 1
            else :
                return False
        for k, v in word.items() :
            if v != 0 : 
                return False
        return True
        